kong/plugins/http-log/handler.lua

```diff
@@ -72,36 +48,47 @@ local function log(premature, conf, body, name)
   end
   name = "[" .. name .. "] "
 
-  local ok, err
+  local headers
+  local req_url
   local parsed_url = parse_url(conf.http_endpoint)
-  local host = parsed_url.host
-  local port = tonumber(parsed_url.port)
 
-  local sock = ngx.socket.tcp()
-  sock:settimeout(conf.timeout)
-
-  ok, err = sock:connect(host, port)
-  if not ok then
-    ngx.log(ngx.ERR, name .. "failed to connect to " .. host .. ":" .. tostring(port) .. ": ", err)
-    return
+  if parsed_url.userinfo then
+    -- for lua-resty-http client Authorization
+    headers = {
+      ["Content-Type"] = conf.content_type,
+      ["Authorization"] = string_format(
+        "Basic %s", ngx.encode_base64(parsed_url.userinfo))
+    }
+    req_url = parsed_url.scheme .. "://" .. parsed_url.host .. ":" .. parsed_url.port
+      .. parsed_url.path .. (parsed_url.params or "")
+  else
```

Comment:
style: please jump a line above this `else` token, as per our updated style guide. Also, the indentation for the above concatenation operation could be improved and made more readable ideally.